File: pinyougou-parent/pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/SeckillGoodsController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbSeckillGoods;
import com.pinyougou.seckill.service.SeckillGoodsService;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/managerSeckill")
public class SeckillGoodsController {

    @Reference
    private SeckillGoodsService seckillGoodsService;

    @RequestMapping("/findAll")
    public List<TbSeckillGoods> findAll(){
        return seckillGoodsService.findAll();
    }

    @RequestMapping("/findCheckedSeckillGoods")
    public List<TbSeckillGoods> findCheckedSeckillGoods(){
        return seckillGoodsService.findCheckedSeckillGoods();
    }

    @RequestMapping("/findOne")
    public TbSeckillGoods findOne(Long id){
        return seckillGoodsService.findOne(id);
    }

    @RequestMapping("/delete")
    public Result dele(Long[] ids){
        try {
            seckillGoodsService.delete(ids);
            return new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(true,"删除失败");
        }
    }

    @RequestMapping("/add")
    public Result add(@RequestBody TbSeckillGoods seckillGoods){
        try {
            seckillGoodsService.add(seckillGoods);
            return new Result(true,"新建成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"新建失败");
        }
    }

    @RequestMapping("/update")
    public Result update(@RequestBody TbSeckillGoods seckillGoods){
        try {
            seckillGoodsService.update(seckillGoods);
            return new Result(true,"修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }

    @RequestMapping("/alterStatus")
    public Result alterStatus(Long id, String status){
        try {
            seckillGoodsService.alterStatus(id,status);
            return new Result(true,"修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"修改失败");
        }
    }
}

File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/controller/seckillGoodsController.js
app.controller('seckillGoodsController',function ($scope,$controller,seckillService) {

    $controller('baseController',{$scope:$scope});

    $scope.search = function () {
        seckillService.findAll().success(
            function (response) {
                $scope.list = response;
            }
        )
    }

    $scope.findOne = function (id) {
        seckillService.findOne(id).success(
            function (response) {
                $scope.entity = response;
            }
        )
    }

    $scope.dele = function () {
        seckillService.dele($scope.selectIds).success(
            function (response) {
                if(response.success){
                    alert(response.message)
                    $scope.search();
                }else {
                    alert(response.message)
                }
            }
        )
    }

    $scope.save = function () {
        var object = null;
        if ($scope.entity.id != null) {
            object = seckillService.update($scope.entity);
        }else {
            object = seckillService.add($scope.entity);
        }
        object.success(
            function (response) {
                if(response.success){
                    alert(response.message)
                    $scope.search();
                }else {
                    alert(response.message)
                }
            }
        )
    }

    $scope.findCheckedSeckillGoods = function () {
        seckillService.findCheckedSeckillGoods().success(
            function (response) {
                $scope.checkedList = response;
            }
        )
    }

    $scope.alterStatus = function () {
        seckillService.alterStatus($scope.selectIds,1).success(
            function (response) {
                if(response.success){
                    alert(response.message)
                    $scope.findCheckedSeckillGoods();
                }else {
                    alert(response.message)
                }
            }
        )
    }

})